src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -612,9 +612,15 @@ private CodeTypeBase CreateModelClasses(OpenApiUrlSpaceNode rootNode, OpenApiUrl
                     });
                 }
                 return unionType;
-            } else if(schema?.Type?.Equals("object") ?? false) {
+            } else if(schema.IsObject()) {
                 // referenced schema, no inheritance or union type
                 return CreateModelClassAndType(rootNode, currentNode, schema, operation, parentElement, codeNamespace);
+            } else if (schema.IsArray())
+            {
+                // collection of referenced schema
+                var type = CreateModelClassAndType(rootNode, currentNode, schema, operation, parentElement, codeNamespace, "");
```

Comment:
Should this have the suffix "Collection" and not be set of CollectionKind Array, or should it by type Pet[] ?

Reply:
we should go with an array of pets, I'd like to avoid generating extra classes when it's not needed. We might have to tweak the deserialization process and/or the executor body during testing but this scenario should mostly already be handled. (besides the generation part)